lib/ansible/module_utils/distro/__init__.py

```diff
@@ -22,7 +22,7 @@
 from __future__ import annotations
 
 # The following makes it easier for us to script updates of the bundled code
-_BUNDLED_METADATA = {"pypi_name": "distro", "version": "1.8.0"}
+_BUNDLED_METADATA = {"pypi_name": "distro", "version": "1.9.0"}
```

Comment:
@DhaneshRaghavan Thanks for the contribution. There are other changes which goes with this change. 

```diff
diff --git a/lib/ansible/module_utils/distro/_distro.py b/lib/ansible/module_utils/distro/_distro.py
index e57d6b6854..fa03e3da36 100644
--- a/lib/ansible/module_utils/distro/_distro.py
+++ b/lib/ansible/module_utils/distro/_distro.py
@@ -1,4 +1,4 @@
-# Copyright 2015,2016,2017 Nir Cohen
+# Copyright 2015-2021 Nir Cohen
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -30,7 +30,6 @@ Python 2.6 and removed in Python 3.8. Still, there are many cases in which
 access to OS distribution information is needed. See `Python issue 1322
 <https://bugs.python.org/issue1322>`_ for more information.
 """
from __future__ import annotations
 
 import argparse
 import json
@@ -59,7 +58,7 @@ except ImportError:
     # Python 3.7
     TypedDict = dict
 
-__version__ = "1.8.0"
+__version__ = "1.9.0"
 
 
 class VersionDict(TypedDict):
@@ -129,7 +128,9 @@ _DISTRO_RELEASE_BASENAME_PATTERN = re.compile(r"(\w+)[-_](release|version)$")
 # Base file names to be looked up for if _UNIXCONFDIR is not readable.
 _DISTRO_RELEASE_BASENAMES = [
     "SuSE-release",
+    "altlinux-release",
     "arch-release",
+    "armbian-release",
     "base-release",
     "centos-release",
     "fedora-release",
@@ -155,6 +156,8 @@ _DISTRO_RELEASE_IGNORE_BASENAMES = (
     "system-release",
     "plesk-release",
     "iredmail-release",
+    "board-release",
+    "ec2_version",
 )
 
 
@@ -247,6 +250,7 @@ def id() -> str:
     "rocky"         Rocky Linux
     "aix"           AIX
     "guix"          Guix System
+    "altlinux"      ALT Linux
     ==============  =========================================
 
     If you have a need to get distros for reliable IDs added into this set,
@@ -905,6 +909,9 @@ class LinuxDistribution:
         elif self.id() == "debian" or "debian" in self.like().split():
             # On Debian-like, add debian_version file content to candidates list.
             versions.append(self._debian_version)
+            if self._distro_release_info.get("id") == "armbian":
+                # On Armbian, add version from armbian-release file to candidates list.
+                versions.append(self._armbian_version)
         version = ""
         if best:
             # This algorithm uses the last version in priority order that has
@@ -995,10 +1002,10 @@ class LinuxDistribution:
 
         For details, see :func:`distro.info`.
         """
-        return dict(
+        return InfoDict(
             id=self.id(),
             version=self.version(pretty, best),
-            version_parts=dict(
+            version_parts=VersionDict(
                 major=self.major_version(best),
                 minor=self.minor_version(best),
                 build_number=self.build_number(best),
@@ -1225,6 +1232,16 @@ class LinuxDistribution:
         except FileNotFoundError:
             return ""
 
+    @cached_property
+    def _armbian_version(self) -> str:
+        try:
+            with open(
+                os.path.join(self.etc_dir, "armbian-release"), encoding="ascii"
+            ) as fp:
+                return self._parse_os_release_content(fp).get("version", "")
+        except FileNotFoundError:
+            return ""
+
     @staticmethod
     def _parse_uname_content(lines: Sequence[str]) -> Dict[str, str]:
         if not lines:
@@ -1302,6 +1319,14 @@ class LinuxDistribution:
         if match is not None:
             distro_info["id"] = match.group(1)
 
+            # Armbian release files are not standard as they start with a
+            # comment. Manually set name if it has not been inferred.
+            if distro_info["id"] == "armbian" and distro_info.get(
+                "name", ""
+            ).startswith("#"):
+                distro_info["name"] = "Armbian"
+                distro_info["version_id"] = self._armbian_version
+
         # CloudLinux < 7: manually enrich info with proper id.
         if "cloudlinux" in distro_info.get("name", "").lower():
             distro_info["id"] = "cloudlinux"
```

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Reply:
FYI we are porting a copy of https://github.com/python-distro/distro/blob/master/src/distro/distro.py to [lib/ansible/module_utils/distro/_distro.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/distro/_distro.py)

Reply:
thanks @Akasurde for the shedding the light. this is my first time in Ansible so a little bit of missing things.

Reply:
No worries. I am also here to help.